pkg/controller/controller_utils.go

```diff
@@ -83,6 +83,8 @@ const (
 	// The number of batches is given by:
 	//      1+floor(log_2(ceil(N/SlowStartInitialBatchSize)))
 	SlowStartInitialBatchSize = 1
+	// When scaling down the pods of a ReplicaSet, prioritize removing the specified pod.
+	SpecifiedToDeleteAnnotation string = "k8s.io/specified-to-delete"
```

Comment:
As far as I understand, this is definitely not a good approach, it's very implicit, a good approach would be to modify the api. However, I think ReplicaSet itself is stateless, there is no need to delete "specific" pods.

Reply:
I agree with @googs1025, replicasets are stateless. For this scenario, better alternatives involve deploying high-priority pods in a different replicaset and adding a priority class to it, or using a Statefulset.